packages/cli/src/commands/generate/schema-from-database/generate_schema_command.ts

```diff
@@ -67,8 +70,13 @@ export class GenerateSchemaCommand
     const connectionUriSecretName = args.connectionUriSecret as string;
     const outputFile = args.out as string;
 
+    const backendIdentifierIsStack = 'stackName' in backendIdentifier;
+    const sanitizedBackendId = backendIdentifierIsStack
+      ? BackendIdentifierConversions.fromStackName(backendIdentifier.stackName)
+      : backendIdentifier;
+
     const connectionUriSecret = await this.secretClient.getSecret(
-      backendIdentifier as BackendIdentifier,
+      sanitizedBackendId as BackendIdentifier,
```

Comment:
We should remove this force cast once we have an instance of `BackendIdentifier`

Reply:
Looks like type system was trying to warn about the problem in original implementation.